contracts/base/escrow/dapps/YVault.sol

```diff
@@ -0,0 +1,134 @@
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
+
+// Interfaces
+import "./YVaultInterface.sol";
+import "../../../providers/yearn/IVault.sol";
+
+/*****************************************************************************************************/
+/**                                             WARNING                                             **/
+/**                      DAPP CONTRACT IS AN EXTENSION OF THE ESCROW CONTRACT                       **/
+/**  ---------------------------------------------------------------------------------------------  **/
+/**  Because there are multiple dApp contracts, and they all extend the Escrow contract that is     **/
+/**  itself upgradeable, they cannot have their own storage variables as they would cause the the   **/
+/**  storage slots to be overwritten on the Escrow proxy contract!                                  **/
+/**                                                                                                 **/
+/**  Visit https://docs.openzeppelin.com/upgrades/2.6/proxies#upgrading-via-the-proxy-pattern for   **/
+/**  more information.                                                                              **/
+/*****************************************************************************************************/
+/**
+    @notice This contract is used to define the yVault dApp actions available. All dapp actions are invoked via
+        delegatecalls from Escrow contract, so this contract's state is really Escrow.
+    @author develop@teller.finance
+ */
+contract YVault is YVaultInterface, BaseEscrowDapp {
+    using AddressLib for address;
+    using Address for address;
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    /** External Functions */
+    /**
+        @notice Deposits the specified amount of the native unwrapped token (same as token() returns) into the Vault
+        @param tokenAddress The address of the token being deposited
+        @param amount The amount of tokens to be deposited into the vault
+     */
+    function deposit(address tokenAddress, uint256 amount) public onlyBorrower {
+        IVault iVault = _getYVault(tokenAddress);
+        IERC20 underlyingToken = IERC20(iVault.underlying());
+        uint256 tokenBalanceBeforeDeposit =
+            underlyingToken.balanceOf(address(this));
+        IERC20(tokenAddress).safeApprove(address(iVault), amount);
+        iVault.deposit(amount);
+        uint256 tokenBalanceAfterDeposit =
+            underlyingToken.balanceOf(address(this));
+        require(
+            tokenBalanceAfterDeposit > tokenBalanceBeforeDeposit,
+            "YEARN_BALANCE_NOT_INCREASED"
+        );
+        _tokenUpdated(tokenAddress);
+
+        emit YearnDeposited(
+            tokenAddress,
+            address(iVault),
+            amount,
+            tokenBalanceAfterDeposit
+        );
```

Comment:
Lets do this: https://github.com/teller-protocol/teller-protocol-v1/pull/252#discussion_r585630761